src/v/transform/transform_manager.cc

```diff
@@ -106,7 +106,9 @@ class processor_table {
           , _probe(std::move(probe)) {}
 
         transform::processor* processor() const { return _processor.get(); }
-        ss::lw_shared_ptr<transform::probe> probe() const { return _probe; }
+        const ss::lw_shared_ptr<transform::probe>& probe() const {
```

Comment:
nit: any reason not to return a const ref to the probe itself then?

Reply:
Because we need to make copies of it when creating new processors.

Reply:
Ah, gotcha